src/comm_mpi.cpp

```diff
@@ -222,17 +226,15 @@ void Comm::exchange() {
   for(phase = 0; phase < 6; phase ++) {
     proc_num_send[phase] = 0;
     proc_num_recv[phase] = 0;
-    Kokkos::deep_copy(pack_ranks_migrate,proc_rank);
 
     T_INT count = 0;
     Kokkos::deep_copy(pack_count,0);
 
     if(proc_grid[phase/2]>1) {
       // If a previous phase resized the AoSoA, export ranks needs to be resized as well
-      if(pack_ranks_migrate.extent(0) != x.size()) {
-        Kokkos::resize(pack_ranks_migrate, x.size());
-        Kokkos::deep_copy(pack_ranks_migrate,proc_rank);
-      }
+      if(pack_ranks_migrate.extent(0) != x.size())
+        Kokkos::realloc(pack_ranks_migrate, x.size());
```

Comment:
I bet we could be smart about this and avoid reallocation by allocating enough memory such that we don't exceed and then just us a subview to get an effective view of size `x.size()`